iothub/device/src/Transport/AmqpIot/AmqpIotSendingLink.cs

```diff
@@ -130,6 +130,13 @@ private async Task<Outcome> SendAmqpMessageAsync(AmqpMessage amqpMessage, Cancel
                         cancellationToken)
                     .ConfigureAwait(false);
             }
+            catch (OperationCanceledException oce) when (!cancellationToken.IsCancellationRequested)
+            {
+                // OperationCanceledException may be thrown here when there is networking disconnect
+                // even if cancellation has not been requested yet. This case is treated as a transient
```

Comment:
Interesting. 
Fyi - Mqtt had a similar behavior that they seem to have fixed in their latest update.